### PR TITLE
Update to support roles.

### LINKED ISSUE #1114: [LUIS CLI] does not support entity labels in utterances with roles
## Tool
Name: LUIS
Version: master
OS: Win10

## Describe the bug
LUIS portal can import and export applications with role labels in utterances. 
LUIS CLI - can export a model with role definition corectly
LUIS CLI - drops role definition in labelled utterances on the floor and imports the model without roles.

## To Reproduce
Import the LUIS model via the LUIS portal - you will see roles for utterances picked up correctly. e.g. "my name is {userName:firstName=vishwac}. 

Now use the CLI to import this and roles are dropped. 

```bash
> luis import application --in <path to json> --authoringKey <auth-key>
```

Model JSON
``` json
{
  "luis_schema_version": "3.2.0",
  "versionId": "0000000002",
  "name": "10.EchoBot(vkannan)-GetUserNameDialog.en-us.lu",
  "desc": "Model for 10.EchoBot app, targetting vkannan for GetUserNameDialog.lu file",
  "culture": "en-us",
  "tokenizerVersion": "1.0.0",
  "intents": [
    {
      "name": "GetUserName"
    },
    {
      "name": "Greeting"
    },
    {
      "name": "None"
    }
  ],
  "entities": [
    {
      "name": "userName",
      "roles": [
        "lastName",
        "firstName"
      ]
    }
  ],
  "composites": [],
  "closedLists": [],
  "patternAnyEntities": [],
  "regex_entities": [],
  "prebuiltEntities": [],
  "model_features": [],
  "regex_features": [],
  "patterns": [],
  "utterances": [
    {
      "text": "hello",
      "intent": "Greeting",
      "entities": []
    },
    {
      "text": "hi",
      "intent": "Greeting",
      "entities": []
    },
    {
      "text": "hiya",
      "intent": "Greeting",
      "entities": []
    },
    {
      "text": "my name is vishwac",
      "intent": "GetUserName",
      "entities": [
        {
          "entity": "userName",
          "role": "firstName",
          "startPos": 11,
          "endPos": 17
        }
      ]
    },
    {
      "text": "my name is vishwac sena kannan",
      "intent": "GetUserName",
      "entities": [
        {
          "entity": "userName",
          "role": "firstName",
          "startPos": 11,
          "endPos": 22
        },
        {
          "entity": "userName",
          "role": "lastName",
          "startPos": 24,
          "endPos": 29
        }
      ]
    },
    {
      "text": "vishwac",
      "intent": "GetUserName",
      "entities": [
        {
          "entity": "userName",
          "role": "firstName",
          "startPos": 0,
          "endPos": 6
        }
      ]
    },
    {
      "text": "vishwac is my name",
      "intent": "GetUserName",
      "entities": [
        {
          "entity": "userName",
          "role": "firstName",
          "startPos": 0,
          "endPos": 6
        }
      ]
    },
    {
      "text": "you can call me vishwac",
      "intent": "GetUserName",
      "entities": [
        {
          "entity": "userName",
          "role": "firstName",
          "startPos": 16,
          "endPos": 22
        }
      ]
    }
  ],
  "settings": []
}
```
## Expected behavior
Roles are the new feature announcement for //build. It needs to work E2E through the CLI

[bug]

### PR DESCRIPTION
**Pull request by chrimc62**
Fixes #1114 #912 #1182 

## Proposed Changes
Update to the latest LUIS swagger file.  This also required moving from the region model to the endpoint model.

### CODE REVIEW

File: packages/LUIS/bin/luis.js [hunk 1]
```diff
@@ -105,6 +105,8 @@ async function runProgram() {
     args.region = args.region || serviceIn.region || config.region || "westus";
     args.cloud = args.cloud || serviceIn.cloud || config.cloud;
     args.customHeaders = { "accept-language": "en-US" };
+    args.endpoint = args.endpoint || serviceIn.endpoint || config.endpoint;
+    args.authoringEndpoint = args.authoringEndpoint || serviceIn.authoringEndpoint || config.authoringEndpoint || `https://${args.region}.api.cognitive.microsoft.com/luis/api/v2.0/`;
```
Comment (tomlm) [inline packages/LUIS/bin/luis.js]: >com [](start = 162, length = 3)

This should be {args.cloud}

Comment (tomlm) [inline packages/LUIS/bin/luis.js]: >authoringEndpoint [](start = 9, length = 17)

no need for authoring endpoint, combo of endpoint + cloud is sufficient.
"endpoint":"westeurope",
"cloud":"eu"

File: packages/LUIS/bin/luis.js [hunk 2]
```diff
@@ -105,6 +105,8 @@ async function runProgram() {
     args.region = args.region || serviceIn.region || config.region || "westus";
     args.cloud = args.cloud || serviceIn.cloud || config.cloud;
     args.customHeaders = { "accept-language": "en-US" };
+    args.endpoint = args.endpoint || serviceIn.endpoint || config.endpoint;
```
Comment (tomlm) [inline packages/LUIS/bin/luis.js]: >endpoint [](start = 9, length = 8)

Why add endpoint?  We had region + cloud

**Comment by vishwacsena** [comment]
Verified that roles work E2E with import and export. 

Full set of commands I tried - 
```
> luis import application
> luis import version
> luis train version
> luis publish version
> luis query
```

File: packages/LUIS/src/luisAuthoringContext.ts [hunk 3]
```diff
@@ -10,14 +10,20 @@ const packageName = "luis-apis";
...
 
   /**
    * Initializes a new instance of the LuisAuthoringContext class.
+   * @param endpoint Supported Cognitive Services endpoints (protocol and hostname, for example:
+   * https://westus.api.cognitive.microsoft.com).
    * @param credentials Subscription credentials which uniquely identify client subscription.
    * @param [options] The parameter options
    */
-  constructor(credentials: msRest.ServiceClientCredentials, options?: msRest.ServiceClientOptions) {
+  constructor(endpoint: string, credentials: msRest.ServiceClientCredentials, options?: msRest.ServiceClientOptions) {
```
Comment (tomlm) [inline packages/LUIS/src/luisAuthoringContext.ts]: >endpoint [](start = 14, length = 8)

this is a breaking change, there should at least be a major version bump

File: packages/LUIS/src/luisAuthoring.ts [hunk 4]
```diff
@@ -25,11 +25,13 @@ class LuisAuthoring extends LuisAuthoringContext {
...
   /**
    * Initializes a new instance of the LuisAuthoring class.
+   * @param endpoint Supported Cognitive Services endpoints (protocol and hostname, for example:
+   * https://westus.api.cognitive.microsoft.com).
    * @param credentials Subscription credentials which uniquely identify client subscription.
    * @param [options] The parameter options
    */
-  constructor(credentials: msRest.ServiceClientCredentials, options?: msRest.ServiceClientOptions) {
-    super(credentials, options);
+  constructor(endpoint: string, credentials: msRest.ServiceClientCredentials, options?: msRest.ServiceClientOptions) {
```
Comment (tomlm) [inline packages/LUIS/src/luisAuthoring.ts]: >endpoint [](start = 14, length = 8)

this is a breaking change, there should be a major version bump

File: packages/LUIS/src/models/parameters.ts [hunk 5]
```diff
@@ -134,6 +91,18 @@ export const domainName: msRest.OperationURLParameter = {
     }
   }
 };
+export const endpoint: msRest.OperationURLParameter = {
```
[the remark this answers is not in the document]
Comment (chrimc62) [reply packages/LUIS/src/models/parameters.ts]: Yes.  I think the real question is if our swagger file is wrong or autorest has a bug.

**Pull request merged**